### PR TITLE
feat: users awaiting approval

### PR DESCRIPTION
**Pull request by IC-1101asterisk**
## Description
Now users can be created without a channel, they are considered "unactivated users" or "User awaiting approval". They need to have their request accepted by an admin to become a normal user (or another admin)

Fixes FL-989

## How to test
Connect with the SSO using an OIDC provider see: https://github.com/Substra/substra-backend/pull/609
This will create an unactivated user thus a request on the User page of any admin.
This is linked to this backend PR: https://github.com/Substra/substra-backend/pull/680
Also I made a POST to use with curl to easily create unactivated users without the OIDC provider (not to be merged in the backend).

## Notes for developers and reviewers:

-   Think to update CHANGELOG.md before merge if needed !

### CODE REVIEW

File: src/App.tsx
Comment (oleobal) [inline src/App.tsx:77]: Ultra nitpick, this might communicate the intent very slightly more clearly.

```suggestion
    if (!channel && !(onLoginPage || onResetPage)) {
```

File: src/api/UsersApi.ts
Comment (oleobal) [inline src/api/UsersApi.ts:53]: According to the [axios documentation](https://axios-http.com/docs/req_config) I think you can replace this with something like


```suggestion
    API.put(
        API_PATHS.USERS_AWAITING_APPROVAL,
        payload,
        {"params": {"username": key}}
    );
```

(I haven't tried it)

File: src/routes/users/useUsersStore.tsx
Comment (Milouu) [inline src/routes/users/useUsersStore.tsx:216]: I don't think all this is necessary as we refetch everything after deletion

**Review by Milouu — DISMISSED** [review]

**Comment by Milouu** [comment]
On every login the page "Waiting for your approval" is very visible, which would be fine on first connexion but i think it's weird to see it everytime 🤔

**Comment by Milouu** [comment]
Capture d’écran 2023-07-24 à 10 23 11
Did you try playing with different orgs ? I have a weird user that i didn't create waiting for approval so i'm wondering if that's suppose to come from another org (as 'external' is generally the username when it's a user from other orgs)

**Review by Milouu — APPROVED** [review]

**Pull request merged**
